test/tests/functional/pbs_mom_walltime.py

```diff
@@ -127,18 +127,36 @@ def test_suspend_time_not_counted_in_walltime(self):
         jid1 = self.server.submit(J1)
         self.server.expect(JOB, {'job_state': 'R'}, id=jid1)
 
+        # Accumulate wall time
+        time.sleep(10)
+
         self.server.sigjob(jobid=jid1, signal="suspend")
         self.server.expect(JOB, {'job_state': 'S'}, id=jid1)
 
+        # Make sure the sched cycle is completed before reading
+        # the walltime
+        self.server.manager(MGR_CMD_SET, MGR_OBJ_SERVER,
+                            {'scheduling': 'True'}, expect=True)
+        self.server.manager(MGR_CMD_SET, MGR_OBJ_SERVER,
+                            {'scheduling': 'False'}, expect=True)
+
+        jstat = self.server.status(JOB, id=jid1,
+                                   attrib=['resources_used.walltime'])
+        walltime = jstat[0]['resources_used.walltime']
+        self.logger.info("Walltime before sleep %s" % walltime)
+        self.server.manager(MGR_CMD_SET, MGR_OBJ_SERVER,
+                            {'scheduling': 'True'})
+
         time.sleep(15)
-        self.server.sigjob(jobid=jid1, signal="resume")
-        self.server.expect(JOB, {'job_state': 'R'}, id=jid1)
 
-        self.server.expect(JOB, {ATTR_state: 'F'}, id=jid1, extend='x',
-                           offset=5)
+        # Used walltime should remain the same
+        self.server.expect(JOB, {'resources_used.walltime': walltime}, op=EQ,
+                           id=jid1, extend='x')
 
-        # Used walltime should be less than the sleep time after suspend
-        self.server.expect(JOB, {'resources_used.walltime': 10}, op=LE,
+        self.server.sigjob(jobid=jid1, signal="resume")
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid1)
+        time.sleep(10)
+        self.server.expect(JOB, {'resources_used.walltime': walltime}, op=GT,
```

Comment:
There are 2 types of validation here in the test: first that job's wall time should not be getting accumulated when it is suspended. Second, job's walltime should not get accumulated in the overall walltime.
you are checking first but not second. for testing second I suggest adding a range like >= 30 (job's sleep time) but <40 (adding 10s buffer since min mom poll time is 10s).